common/djangoapps/student/models.py

```diff
@@ -1569,7 +1569,11 @@ def unenroll_by_email(cls, email, course_id):
         """
         try:
             user = User.objects.get(email=email)
-            return cls.unenroll(user, course_id)
+            record = cls.objects.get(user=user, course_id=course_id)
+            if record.refundable():
```

Comment:
This check also exists before submitting the refund, why do we need to add it here also?

https://github.com/edx/edx-platform/blob/64555c60c043acb00ff68f508fad73d09691fc12/lms/djangoapps/commerce/signals.py#L29-L29